planner/core/find_best_task.go

```diff
@@ -484,31 +605,42 @@ func (ds *DataSource) findBestTask(prop *property.PhysicalProperty) (t task, err
 
 	t, err = ds.tryToGetDualTask()
 	if err != nil || t != nil {
-		return t, err
+		clock.Dec(1)
+		return t, 1, err
 	}
 
 	t = invalidTask
 	candidates := ds.skylinePruning(prop)
 
+	cntPlan = 0
 	for _, candidate := range candidates {
 		path := candidate.path
 		if path.PartialIndexPaths != nil {
 			idxMergeTask, err := ds.convertToIndexMergeScan(prop, candidate)
 			if err != nil {
-				return nil, err
+				return nil, 0, err
+			}
+			if !idxMergeTask.invalid() {
```

Comment:
Is it possible to move this check its father like `if the child is invalid, restore the clock`? Will this make the code cleaner?

Reply:
No, it will affect the returned value `planCnt`, which is used as an important indicator during `childTask` combination.